crates/wasmparser/src/validator/operators.rs

```diff
@@ -676,16 +677,17 @@ impl OperatorValidator {
                 Some(ValType::Ref(RefType { nullable: _, heap_type: HeapType::Index(z) })) => {
                     let ctft2 = func_type_at(resources, cont_type_at(resources, z)?)?;
                     // Now we must check that (ts2' -> ts2) <: $ft
+                    // This method should be exposed by resources to make this correct
                     for (tagty, ct2ty) in tagtype.outputs().zip(ctft2.inputs()) {
-                        // Note: function subtyping is contravariant in input types.
+                        // Note: according to spec we should check for equality here
```

Comment:
So, this comes from the function references spec text:
```
Note: Function types are invariant for now. This may be relaxed in future extensions.
```
Meaning, we should just perform equality pointwise on the function types.